File: leetcode/js/41.Missing Positive.js
var findMissingPositive = function(A){
    let n = A.length
    let present = []
    for(let i=0;i<n;i++){
     if(A[i]>0&&A[i]<n){
       present[A[i]] = true
     }
    }
    for(let i=1;i<n+1;i++){
     if(!present[i]){
       return i
     }
    }
    return n+1
 }
 
 const nums = [0,-1,3,1]
 console.log(findMissingPositive(nums))

File: leetcode/python/41.Missing Positive.py
def findMissingPositive(A):
  n = len(A)
  present = [False for i in range(n)]
  for i in range(n):
    if A[i] > 0 and A[i] <= n:
      present[A[i]] = True
  for i in range(1,n+1):
    if not present[i]:
      return i
  return n+1

nums=[0,-1,3,1]
print(findMissingPositive(nums))